pydantic_ai_slim/pydantic_ai/_acp.py

```
@@ -0,0 +1,833 @@
+
+    def _validate_path(self, path_str: str) -> Path:
+        """Securely resolve path against root_dir."""
+        try:
+            path = Path(path_str)
+            if not path.is_absolute():
+                path = (self.root_dir / path).resolve()
+            else:
+                path = path.resolve()
+
+            if not str(path).startswith(str(self.root_dir)):
+                raise ValueError('Path outside root directory')
```

Ruff (0.15.2)

[warning] Abstract `raise` to an inner function

(TRY301)